worlds/dark_souls_3/__init__.py

```diff
@@ -0,0 +1,281 @@
+# world/dark_souls_3/__init__.py
+import json
+import os
+
+from random import randint
+from .Options import dark_souls_options  # the options we defined earlier
+from .Items import DarkSouls3Item  # data used below to add items to the World
+from .Locations import DarkSouls3Location  # same as above
+from .data.items_data import weapons_upgrade_5_table, weapons_upgrade_10_table, item_dictionary_table, key_items_list
+from .data.locations_data import dictionary_table, cemetery_of_ash_table, fire_link_shrine_table, \
+    high_wall_of_lothric, \
+    undead_settlement_table, road_of_sacrifice_table, consumed_king_garden_table, cathedral_of_the_deep_table, \
+    farron_keep_table, catacombs_of_carthus_table, smouldering_lake_table, irithyll_of_the_boreal_valley_table, \
+    irithyll_dungeon_table, profaned_capital_table, anor_londo_table, lothric_castle_table, grand_archives_table, \
+    untended_graves_table, archdragon_peak_table, firelink_shrine_bell_tower_table
+from ..AutoWorld import World, WebWorld
+from BaseClasses import MultiWorld, Location, Region, Item, RegionType, Entrance, Tutorial, ItemClassification
+from ..generic.Rules import set_rule
+
+
+class DarkSouls3Web(WebWorld):
+    tutorials = [Tutorial(
+        "Multiworld Setup Tutorial",
+        "A guide to setting up the Archipelago Dark Souls 3 randomizer on your computer.",
+        "English",
+        "setup_en.md",
+        "setup/en",
+        ["Marech"]
+    )]
+
+
+class DarkSouls3World(World):
+    """
+    Dark souls 3 is an Action role-playing game and is part of the Souls series developed by FromSoftware.
+    Played in a third-person perspective, players have access to various weapons, armour, magic, and consumables that
+    they can use to fight their enemies.
+    """
+
+    def __init__(self, world: MultiWorld, player: int):
+        super().__init__(world, player)
+        self.locked_items = []
+        self.locked_locations = []
+        self.main_path_locations = []
+
+    def create_item(self, name: str) -> Item:
+        data = self.item_name_to_id[name]
+        return DarkSouls3Item(
+            name, ItemClassification.progression if name in key_items_list else
+            ItemClassification.useful, data, self.player)
```

Comment:
This causes generation failures as you have more items than locations with the default settings and none are marked as junk so none can be discarded. You should have a minimal number of progression and useful items so other people in the multiworld can get items too. Items and location counts should also never be unequal.

Reply:
Fixed. I now use 3 differents classifications (progression, useful and filler) depending on the item category.